### DISCUSSION TITLE
Flashing Seeed XIAO ESP32-S3 with tasmota32s3.bin (or tasmota32s3.factory.bin)

### DISCUSSION THREAD

**Discussion by NinoBenci** [arendst/Tasmota#21539] [Support request] [answered]
I've been using a few XIAO ESP32-S3 for various projects lately and was hoping to implement one project using the Tasmota firmware and a XIAO ESP32-S3. I downloaded the lasted version specifically for the ESP32-S3 and have tried to use Chrome via
https://tasmota.github.io/install/
Unfortunately, I have not been successful in getting the binary flashed, no matter what I try. Has anyone else had any issues?
Yes, the port is recognized as COM13. I get as far as the flashing process indicating that it was completed, with and without erasing the device, then nothing. Manually resetting the device does nothing. No WiFi hot spot is shown when I do a scan.
Any suggestions would be welcomed.
Thanks.

**Comment by sfromis** [comment]
I'd recommend to not do any download first (you may not have downloaded the correct file with `.factory` in the name), but let the Tasmota Web Installer choose the chip architecture automatically, with you only needing to select the serial port.

I'd expect the standard install setup to work fully, including PSRAM.

**Comment by NinoBenci** [comment]
I have tried noting your comments. After connecting (USB/JTAG Connection on COM13), and selecting the first option, Tasmota...
The process indicates completion, then when I select "NEXT", the response I get is Device Dashboard asking to "INSTALL TASMOTA (ENGLISH)" or "LOGS & CONSOLE". Even doing a hard reset, there is no response, no WiFi spot is created, nor anything on the console.
I have even tried using the ESP-Flasher tool. although V2.2.4 returns a Virus positive error on downloading it, thus I can only use V2.2.3.
At this point, I'm at a dead end.

**Comment by NinoBenci** [comment]
Further tests... Using ESP-Flasher V2.2.1. The only version I gan get to reliably work.

Testing with Initial Firmware binaries, NOT OTA Binaries. The flashing process completes but I get no response from the chip. I'm starting to consider that I have a faulty chip... Although , programming a basic web app from my test code (Arduino IDE V2.3.2) seems to confirm that the chip works and is fully functional.

Cheers.

**Reply by sfromis** [reply]
Well, you did not tell what **exactly** you tried doing, like what **specific** binary you are trying to flash, and exactly what you do. As a starting point, use `tasmota32s3.factory.bin`

**Reply by NinoBenci** [reply]
As there are only two possible options, excluding the OTA binaries, I have attempted with the following "**Factory binaries to be used for initial flashing using esptool**" binaries, downloaded from
"**https://ota.tasmota.com/tasmota32/release/**"
I have used the "Webinstaller" and have tried "ESP_Flasher V2.2.3 as V2.2.4 is marked as having a virus by Firefox, so I'm not planning on using that version. I will try using Esptool.py from Espressif later today.

The binaries that I have used were;

tasmota32s3.factory.bin | 2852k | 20240514 16:12
no success...
Then I tried,
tasmota32.factory.bin | 2913k | 20240514 16:11
no success...

I hope this clarifies which binaries were attempted.

![capture_004_03062024_093545](https://github.com/arendst/Tasmota/assets/5866774/3223c955-db81-4baa-80d8-80e8593fd7d3)

**Reply by sfromis** [reply]
When you have a board with ESP32-S3, the Tasmota binary **must** have a name starting with `tasmota32s3`, all others cannot work.

One issue which can lead to trouble is if your USB setup cannot provide the power for the ESP board to get on Wifi. That is not a rare thing.....

You also need to make sure to put the board in flashing mode. With ESP32-S3 boards, the usual way is to hold the "boot button", while clicking the "reset" button before releasing "boot" again.

Note that when that Firefox virus thing yells "VIRUS!" for downloads with Python code + runtime, that is pure BS. Sure, that type of tools are hard to deal with for a virus scanner, just goes to illustrate how bad they are. Certainly not trustworthy, either way.

**Reply by NinoBenci** [reply]
Hi.
That is exactly what I have tried, with no success.
The XIAO ESP32-S3 I can program and run off the same powered USB-C (measured at 5.1V/2A max load) port with the Arduino IDE and several examples with no issues. All tested Seeed and Arduino examples work.
Whether Firefox is correct or not, that is another issue to deal with on a better day.
I am currently using Chrome...

**Comment by arendst** [comment]
I ordered the sense and will see how to make tasmota work.

**Reply by NinoBenci** [reply]
UPDATE:

Using the latest bin firmware (14.1.0 Rachel on ESP32), and bin file: tasmota32s3.factory.bin | 2871k | 20240603 14:27

I have managed to program the XIAO ESP32-S3 (not the sense version) using Esp-Flasher 2.2.1
The tricky part is getting the module into BOOT programming mode.
Disconnect from USB-C, press and hold the the BOOT button, plug in the USB-C cable, release the BOOT button, then briefly press the RESET button... Yeah, its a fiddle.

After programming and a reset, the following is shown on the log screen,

Leaving...
Hard Resetting...
Hard resetting via RTS pin...
Done! Flashing is complete!

Showing logs:
[21:56:54]00:00:00.542 BRY: Trying to expand FS by 4096 kB
[21:56:54]00:00:00.711 BRY: Successfully resized FS, restarting
[21:56:54]00:00:00.730 BRY: Berry initialized, RAM used 5200 bytes
[21:56:54]00:00:00.747 Project tasmota - Tasmota Version 14.1.0(release-tasmota32)-3_0_0(2024-06-03T11:56:29)
[21:56:55]00:00:01.026 WIF: WifiManager active for 3 minutes
[21:56:55]00:00:01.698 HTP: Web server active on tasmota-91AE84-3716 with IP address 192.168.4.1
[21:56:56]00:00:02.654 APP: Restarting
[21:56:57]ESP-ROM:esp32s3-20210327
[21:56:57]Build:Mar 27 2021
[21:56:57]rst:0xc (RTC_SW_CPU_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
[21:56:57]Saved PC:0x40374f2f
[21:56:57]SPIWP:0xee
[21:56:57]mode:DIO, clock div:1
[21:56:57]load:0x3fce3818,len:0x158
[21:56:57]load:0x403c9700,len:0x4
[21:56:57]load:0x403c9704,len:0xa88
[21:56:57]load:0x403cc700,len:0x2834
[21:56:57]SHA-256 comparison failed:
[21:56:57]Calculated: 19e7ee89828b1c44c0aabce4ac1f0048d571244b3f31c34f5c59bec864bb674c
[21:56:57]Expected: 16d76af4b9da340561882235212dc9c65ed709b98db7905325885de83846471a
[21:56:57]Attempting to boot anyway...
[21:56:57]entry 0x403c9868
[21:56:57]
[21:56:57]00:00:00.002 CMD: Using USB CDC
[21:56:57]00:00:00.002 HDW: ESP32-S3 v0.1 
[21:56:57]00:00:00.015 UFS: FlashFS mounted with 4404 kB free
[21:56:57]00:00:00.026 CFG: Loaded from File, Count 2
[21:56:57]00:00:00.028 SER: Set to 8N1 115200 bit/s
[21:56:57]00:00:00.028 SER: HWCDC supports 115200 bit/s only
[21:56:57]00:00:00.029 FRC: Some settings have been reset (2)
[21:56:57]00:00:00.466 BRY: Berry initialized, RAM used 5200 bytes
[21:56:57]00:00:00.482 Project tasmota - Tasmota Version 14.1.0(release-tasmota32)-3_0_0(2024-06-03T11:56:29)
[21:56:58]00:00:00.889 WIF: WifiManager active for 3 minutes
[21:56:58]00:00:01.527 HTP: Web server active on tasmota-91AE84-3716 with IP address 192.168.4.1

Unfortunately, no WiFi connection is found. Oh well. Further than yesterday. Note that SHA-256 comparison failed.

Cheers.

**Reply by sfromis** [reply]
The log shows that Tasmota has opened the AP for WifiManager, for you to connect to. Some boards have been seen with a bad antenna design resulting in Wifi trouble.....

However, assuming that the first time stamp is from your terminal program, the expected 3 minutes between reboots in AP mode has not passed. **If** that happens very fast, again and again, that would be a typical symptom of lacking power. **If** it was only once, no problem, as Tasmota already reported it going to restart after setting up the file system.

The SHA-256 failure is not important, is in fact just normal when flashing a binary not marked with a hash value.

**Reply by arendst** [reply]
My XiaoESP32S3 just arrived.

- Had to connect the external antenna first.
- Using Tasmota installer from chrome installs Tasmota 4.1.0 release without problem after letting ot do the Erase cycle.
- It then starts the WifiManager.
- Using the serial console I configured my SSID and Password and since then it tries to connect to my AP without success. 

The fact that is fails to connect sounds like you not being able to see a Tasmota AP. The hardware just doesn't work nice.

I'll have a look how to force a connection now ............

NB. Considering the flaky external antenna I don't think I'll be using these devices a lot.

EDIT: Connect failue was due to bad wifi password I entered. It now connects as expected.

**Reply by Jason2866** [reply]
Antenna could be swapped against a better mechanical AP antenna.

**Reply by NinoBenci** [reply]
Thanks for the feedback.
Yes, I agree, the XIAO S3 modules seem to be a bit flaky, although, I have used several in embedded configurations (Arduino Webthings) with only a few annoyances. It's just that I have been unsuccessful in getting Tasmota installed and configured. I have no spares to compare against, only one unit left.
I have also swapped cables several times, although all cables seem to work fine programming via the Arduino IDE. Using a powered HUB with the Arduino IDE.

**Comment by Jason2866** [comment, marked as answer]
Blame espressif for the mess of not being able to put the S3 (S2 too) easy in flash mode. It is a hardware design flaw of the CDC port. This problem can't be fixed from Tasmota.

There is the possibility that the device has preloaded UF2 bootloader. This could makes thing worse to get device in flash mode. For flashing Tasmota the device should be completely erased (esptool.py) when trying to flash Tasmota fails.

If you have the variant with an external antenna, the antenna needs to be installed! Without no wifi will work.

And the device documentation is really bad. No mention of which type of flash and PSRAM is used.
This is mandatory to get PSRAM working.

Edit. Looking in [Arduino Core PR](https://github.com/espressif/arduino-esp32/pull/7971) for support of the device it looks like it uses QIO Flash and OPI PSRAM. So the best precompiled matching Tasmota firmware is
`tasmota32s3-qio_opi.bin` available here https://github.com/tasmota/install/tree/firmware/firmware/unofficial
or via Tasmota WebInstaller selecting in section "unofficial"
<img width="876" alt="image" src="https://github.com/arendst/Tasmota/assets/24528715/ce22da77-7386-4e7f-b6b6-55cf7413aedb">

**Reply by arendst** [reply]
Considering the fact that M5 also mentionend you might need to press the reset button while applying power to get the S3 in boot mode AND that I had no problem flashing it without button press it might well be a USB cable issue too.

**Reply by NinoBenci** [reply]
Will test this tonight.

**Reply by NinoBenci** [reply]
Ok. Some success... Using `tasmota32s3-qio_opi.bin`
Followed your notes and managed to flash the latest bin file.
I still had issues getting into a stable BOOT mode.
Once BIN file was flashed, and SSid and Passsword were configured, the board reset itself 4 times before it found the WIFi signal and presented the WebUI. After that it was stable.
Thanks for the assistance.
PS: I am using the patch antenna that was supplied, no problems.

**Reply by shpegel** [reply]
Thank you jason2866 for getting me on the correct path to getting my xiao ESP32S3 sense working with Tasmota(15.0.1.3 (efb5ab1-lvgl-haspmota). I  wanted note for others who may come this way that I had to do another step to get it to work. After following the steps above and entering my wifi information. Camera restarted, now on my router with its own ip.  I still had no picture but did have the Tasmota "Main" screen menu and 0 FPS. I then went to Main/Configuration/ Auto-Config and chose Xiao-S3-Sense and camera restarted and I had a  picture. 

For flashing with Tasmota Web Flasher I must first hold down the Reset button then hold down the Boot Button (both are down).  Next release the reset button and then release the boot button (after xiao connects). Works 100% of the time.

I will tell you that getting this camera/mcu to work was a major chore and I almost gave up several times. I did get it working  with EspHome, and I will be happy to share the YAML file that finally worked.  I was unaware that the camera I was using was defective. It would work for a while, get noisy and then shutdown the whole thing. I erroneously thought it was the xiao but when the the camera module completely died, I figured it out. Throw in the countless times Arduino IDE failed to compile various sketches for  a variety of reasons and you can see why I was getting frustrated.  

Hang in there, It does work well(now):).
